src/Sylius/Bundle/CoreBundle/EventListener/TaxonDeletionListener.php

```diff
@@ -27,11 +28,15 @@ final class TaxonDeletionListener
     private array $ruleUpdaters;
 
     public function __construct(
-        private SessionInterface $session,
+        private SessionInterface|RequestStack $requestStackOrSession,
         private ChannelRepositoryInterface $channelRepository,
         TaxonAwareRuleUpdaterInterface ...$ruleUpdaters
     ) {
         $this->ruleUpdaters = $ruleUpdaters;
+
+        if ($requestStackOrSession instanceof SessionInterface) {
+            @trigger_error(sprintf('Passing an instance of %s as constructor argument for %s is deprecated as of Sylius 1.12 and will be removed in 2.0. Pass an instance of %s instead.', SessionInterface::class, self::class, RequestStack::class), \E_USER_DEPRECATED);
```

Comment:
What about using https://github.com/symfony/symfony/blob/6.2/src/Symfony/Contracts/Deprecation/function.php#L23?

Reply:
Thanks, I did it in this PR => https://github.com/Sylius/Sylius/pull/14136